src/pkg/packager/deploy.go

```diff
@@ -351,6 +351,13 @@ func (d *deployer) deployComponent(ctx context.Context, pkgLayout *layout.Packag
 	}
 
 	applicationTemplates, err := template.GetZarfTemplates(ctx, component.Name, d.s)
+
+	// This variable needs to be set from the package metadata architecture
+	// This might be able to be expanded to include all the .metadata settings
+	applicationTemplates["###ZARF_PKG_ARCHITECTURE###"] = &variables.TextTemplate{
+		Value: pkgLayout.Pkg.Metadata.Architecture,
```

Comment:
Can you replace all the uses of `pkgLayout.Pkg.Metadata.Architecture` with `pkgLayout.Pkg.Build.Architecture`

Reply:
It looks like we always make them equal, at least we do now, but I think it also just reads a bit better. We know this is set by the system and not the user since it's on deploy.